test/integration/targets/aws_step_functions_state_machine/tasks/main.yml

```diff
@@ -130,6 +130,74 @@
           - update_output.changed == True
           - update_output.state_machine_arn == creation_output.state_machine_arn
 
+    - name: Start execution of state machine -- check_mode
+      aws_step_functions_state_machine_execution:
+        name: "{{ execution_name }}"
+        execution_input: "{}"
+        state_machine_arn: "{{ creation_output.state_machine_arn }}"
+        <<: *aws_connection_info
```

Comment:
Niggle 2:

The common AWS connection info can be better written by adding module_defaults to the top level block:

```
- name: Integration test for AWS Step Function state machine module
  module_defaults:
    group/aws:
      aws_access_key: "{{ aws_access_key }}"
      aws_secret_key: "{{ aws_secret_key }}"
      security_token: "{{ security_token | default(omit) }}"
      region: "{{ aws_region }}"
  block:
```

This means you can then remove all of the ``<<: *aws_connection_info`` lines and the "Set connection information for all tasks" task.

Reply:
Oh, and related niggle 3, please add the new module to lib/ansible/config/module_defaults.yml

Reply:
done. Thanks for the review.